cou/steps/plan.py

```diff
@@ -58,16 +57,16 @@ def normal(text: str) -> str:
     )
 
 
-def apply_plan(upgrade_plan: Any) -> None:
+async def apply_plan(upgrade_plan: UpgradeStep) -> None:
     """Apply the plan for upgrade."""
     result = "X"
     while result.casefold() not in AVAILABLE_OPTIONS:
         result = input(prompt(upgrade_plan.description)).casefold()
         match result:
             case "c":
-                upgrade_plan.run()
+                await upgrade_plan.run()
                 for sub_step in upgrade_plan.sub_steps:
-                    apply_plan(sub_step)
+                    await apply_plan(sub_step)
             case "a":
                 logging.info("Aborning plan")
```

Comment:
Typo:
```suggestion
                logging.info("Aborting plan")
```
Though you could also [consider switching to a different verb](https://inclusivenaming.org/word-lists/tier-1/abort/) altogether, for example 'halt'.

Reply:
done